sentry_sdk/integrations/sanic.py

```diff
@@ -180,21 +183,33 @@ async def _hub_enter(request):
         scope.clear_breadcrumbs()
         scope.add_event_processor(_make_request_processor(weak_request))
 
+    transaction = continue_trace(
+        dict(request.headers),
+        op="http.server",
+        name=request.server_path,
+        source=TRANSACTION_SOURCE_COMPONENT,
+    )
+    request.ctx._sentry_transaction = request.ctx._sentry_hub.start_transaction(
+        transaction
+    ).__enter__()
+
 
-async def _hub_exit(request, **_):
-    # type: (Request, **Any) -> None
+async def _hub_exit(request, response):
+    # type: (Request, BaseHTTPResponse) -> None
+    # TODO: Should these lines go in a try block in case the context got modified?
+    request.ctx._sentry_transaction.set_http_status(response.status)
+    request.ctx._sentry_transaction.__exit__(None, None, None)
     request.ctx._sentry_hub.__exit__(None, None, None)
 
 
-async def _set_transaction(request, route, **kwargs):
+async def _set_transaction(request, route, **_):
     # type: (Request, Route, **Any) -> None
     hub = Hub.current
     if hub.get_integration(SanicIntegration) is not None:
         with capture_internal_exceptions():
             with hub.configure_scope() as scope:
-                route_name = route.name.replace(request.app.name, "").strip(".")
                 scope.set_transaction_name(
-                    route_name, source=TRANSACTION_SOURCE_COMPONENT
+                    "/%s" % route.path, source=TRANSACTION_SOURCE_COMPONENT
```

Comment:
What is in `route.path`? Because we should make sure that transaction names have low cardinality. (So `/user/1231231` would be not good, but `/users/{id}` or similar would be good. (this is important for grouping events together on the backend.

Reply:
Also if the transaction name is a route `/users/{id}` the source should be `TRANSACTION_SOURCE_ROUTE`. See https://develop.sentry.dev/sdk/event-payloads/transaction/#transaction-annotations

Reply:
also applicable above in the `continue_trace` call.

Reply:
The `route.path` is similar to what you wrote in your example, but for Sanic, it also includes the type of the URL parameter. So, for your example, the path would likely be `/users/{id:int}`

Reply:
I would not add a `/`. Because the user will know what is meant anyhow, and maybe his app runs on `/api/user/...` and then `/user/...` would be wrong.